Add Initial name & number Registry Requirements

stevenhartley opened the pull request:
In order to ensure that there is no overlap of assignment of names and/or numbers of uEs, this commit will add in basic rules for reserving names/numbers for specific purposes.

#67

sophokles73 commented on up-l3/README.adoc:
```diff
@@ -64,3 +64,35 @@ This special purpose built uEs are required to implement uProtocol and must be p
+
+== Name & Number Registry
+
+In the following section we will elaborate on requirements for assigning names and numbers (ids) to uEs to avoid name space and number collisions.
```
```suggestion
uProtocol URIs (UUri) contain a uEntity Identifier (`ue_id`) to refer to a particular service _type_. The `ue_id` is a 32 bit unsigned integer of which the least significant 16 bit are used to indicate the service type.

The following section defines sub-ranges of the 16 bit service type address space to be used for different purposes. This will make sure that custom service types defined in a private scope can co-exist with standard uProtocol service types without ambiguities and interference.
```

sophokles73 commented on up-l3/README.adoc:
```diff
@@ -64,3 +64,35 @@ This special purpose built uEs are required to implement uProtocol and must be p
+
+In the following section we will elaborate on requirements for assigning names and numbers (ids) to uEs to avoid name space and number collisions.
+
+=== Reserved uE IDs
```
```suggestion
=== uEntity ID Ranges
```

sophokles73 commented on up-l3/README.adoc:
```diff
@@ -64,3 +64,35 @@ This special purpose built uEs are required to implement uProtocol and must be p
+.Reserved ue_ids
+[width="75%",cols="25%,75%",options="header",]
+|===
+|`ue_id` |Requirement
```
```suggestion
| ID Range | Purpose
```

sophokles73 commented on up-l3/README.adoc:
```diff
@@ -64,3 +64,35 @@ This special purpose built uEs are required to implement uProtocol and must be p
+|===
+|`ue_id` |Requirement
+
+| *`0x0`* | uSubscription
+| *`0x1`* | uDiscovery Client APIs
+| *`0x2`* | uDiscovery Replication APIs
+| *`0x3`* | uTwin
+| *`0x4`* | uStreamer
+| *`0x5-0xFF`* | Reserved for Eclipse-uProtocol
+| *`0x100 - 0xFFF`* | Reserved for link:https://covesa.global/project/uservices/[COVESA-uServices]
+| *`0x1000 - 0x7FFF`* | Reserved for future use
+| *`0x8000 - 0xFFFE`* | Vendor specific
+| *`0xFFFF`* | Reserved
```
```suggestion
| *`0x0000 - 0x00FF`* | Eclipse uProtocol services
| *`0x0100 - 0x0FFF`* |  link:https://covesa.global/project/uservices/[COVESA uServices]
| *`0x1000 - 0x7FFF`* | Future use
| *`0x8000 - 0xFFFE`* | Vendor specific services
| *`0xFFFF`* | Reserved
```

sophokles73 reviewed (commented):
FMPOV we should add the (already) assigned uEntity (type) IDs to the  _Core Services_ section where we already enumerate/describe uProtocol's standard services ...

stevenhartley commented:
> s standard serv

@sophokles73 , I incorporated your changes and also added ustreamer.proto to define the ue_id there, also changed the utwin id to be sequential (3 in lieu of 26).

sophokles73 reviewed (approved):
Very nice :+1: 

LGTM
